Added WRITE Support

rafichris opened the pull request:
Hi Bouni,
I've added write functionality (INPUT MESSAGE: return [{payload: '0', topic: 'Betriebsart/Heizkreis'}]; ) and a lightweight debug capability (INPUT MESSAGE: return [{payload: '', topic: 'INFO/INFO'}]; )

It seems that the settings part is not sent correctly by the SW. (see line 113)
You can also see the missing 'rootname' if you use an input message with topic: 'INFO/INFO'.

GERMAN/ENGLISH may lead to some inconsistent output structure, but this should not be an major issue. 
e.g. 'Access' items are not skipped and stored in "Einstellungen".

I know, you're not maintaining the code, but it would be very kind of you if you're adding the updates.

BR Chris

rafichris committed:
Update luxtronik2-ws.js

Added write support. Input data format example:
   payload: '4'
   topic: 'Betriebsart/Heizkreis'

rafichris committed:
Update luxtronik2-ws.js

Minor Bugfix, added debug functionality if input message is "return [{payload: '', topic: 'INFO/INFO'}];"

rafichris committed:
Update luxtronik2-ws.js

Bouni commented:
Hi @rafichris 

For what ever reason I completely missed this PRfor over 2 years 🤯 

Just a few questions before I mege this:

You changed the comparison of `name` in line 95 to `Zugang: Benutzer` instead of `Access: User` and added some more german comparisons:

https://github.com/Bouni/node-red-contrib-luxtronik2-ws/pull/10/files#diff-065fb345eaa0659734f94a23873079f15a772cd7b33a3e7c9f11fca6b3331cecR95

AS far a I remember and understand the code (haven't used itr in years) these are just there to not process the datapoint and just show "Skipping: Zugang: Benutzer" for xample, right?

If thats the case we ideally woud add the German strings and keep the english ones.

Even better we could maintan an array of strings to ignore which then could be checked and every entry that matches shows "Skipping: <entry>"

I know this comes really late but mybe its worth the thoughts 😄

Nexus1212 commented:
Hi,
write support is a great idea. 
How is the state of this commit?

Best regards
Nexus

rafichris commented:
Hi @Bouni,

an array of strings with multi language support for skipping would be for sure the best solution.

I'm not sure, but i think the "Settings" (German: "Einstellungen") does not contain anything meaningful in json.Content.name
--> A very simple solution is to just use:
`var rootname = "Einstellungen"`
as fallback. (If I remeber correctly)
My proposal: 
Stick to Englisch and use
`var rootname = "Settings"`

Data Debugging:
For testing purpose I've added a raw ouput:
```
 // Output XML structure with RAW values
if(node.topic_parent.includes('INFO') || node.topic_child.includes('INFO')){
    msg.payload['INFO - '+rootname] = json.Content;
}
```
![image](https://user-images.githubusercontent.com/490975/232894148-76a7882e-f27a-40b8-892a-b6c362270958.png)
--> I think you can skip this for the merge.


@Nexus1212: Write support worked very well in the past.
![image](https://user-images.githubusercontent.com/490975/232894900-fab2b617-3b29-4584-864b-677f8ecc994e.png)

I think, I discovered the response via wireshark. (Login to the web interface http://IP_TO_YOUR_HEAT_PUMP/Webserver/index.html, prepare everythink for changing warm water, cooling, etc. - start recording in wireshark - set setting- stop wireshark)
Unfortunately, I can just provide the GERMAN settings:
Cooling: Betriebsart/Kühlung
Warm Water: Betriebsart/Warmwasser
Heating: Betriebsart/Heizkreis

Values:
0 ... Auto
1 ... ZWE (Zweit-Wärme-Erzeugung) / 2nd heating source
2 ... Party
3 ... Ferien / Holidays
4 ... Aus / Off

BR Chris

rafichris committed:
Update luxtronik2-ws.js

rafichris committed:
Update luxtronik2-ws.js

rafichris committed:
Update luxtronik2-ws.js

pulled from https://github.com/BGCDE/node-red-contrib-luxtronik2-ws/commit/25181d05594205623e7a701a8455fe4aab859e87
